packages/hive/src/Adaptor.js

```diff
@@ -0,0 +1,154 @@
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+import { expandReferences } from '@openfn/language-common/util';
+
+import hive from 'hive-driver';
+
+const { TCLIService, TCLIService_types } = hive.thrift;
+const utils = new hive.HiveUtils(TCLIService_types);
+
+const logger = message => console.log(message);
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute` to make working with this API easier.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      connect,
+      ...operations,
+      cleanupState
+    )({
+      ...initialState,
+      ...state,
+    });
+  };
+}
+
+function connect(state) {
+  const { host, port, username, password } = state.configuration;
+
+  const client = new hive.HiveClient(TCLIService, TCLIService_types);
+
+  const connection = new hive.connections.TcpConnection();
+  const auth =
+    username && password
+      ? new hive.auth.PlainTcpAuthentication({
+          username,
+          password,
+        })
+      : new hive.auth.NoSaslAuthentication();
+
+  const connect = client.connect({ host, port }, connection, auth);
+
+  logger('Connect client');
+  return { ...state, connect };
+}
+
+function disconnect(state) {
+  return state.connect.then(async client => {
```

Comment:
This looks wrong - we shouldn't wait for the `connect` promise to resolve here.

See other comments, but once the client (and session ?) have been close, this should just be `await client.close`.

Also I don't think you need `disconnect` AND `cleanupState` - it's a bit confusing to have both. Let's just keep `disconnect` so that our execute looks like this:
```
return commonExecute(
      connect,
      ...operations,
      cleanupState
    )
```